File: portal/src/api/audit.ts
import { get } from "./client";

export interface AuditEntry {
  id: number;
  timestamp: string;
  userEmail: string;
  action: string;
  resourceType: string;
  resourceId: string | null;
  teamSlug: string | null;
  requestBody: Record<string, unknown> | null;
  responseStatus: number | null;
}

export interface AuditLogList {
  entries: AuditEntry[];
  total: number;
  limit: number;
  offset: number;
}

export async function getAuditLog(params?: {
  teamSlug?: string;
  userEmail?: string;
  action?: string;
  limit?: number;
  offset?: number;
}): Promise<AuditLogList> {
  const searchParams = new URLSearchParams();
  if (params?.teamSlug) searchParams.set("team_slug", params.teamSlug);
  if (params?.userEmail) searchParams.set("user_email", params.userEmail);
  if (params?.action) searchParams.set("action", params.action);
  if (params?.limit) searchParams.set("limit", String(params.limit));
  if (params?.offset) searchParams.set("offset", String(params.offset));
  const qs = searchParams.toString();
  return get<AuditLogList>(`/v1/audit${qs ? `?${qs}` : ""}`);
}

export async function getTeamAuditLog(slug: string, params?: {
  limit?: number;
  offset?: number;
}): Promise<AuditLogList> {
  const searchParams = new URLSearchParams();
  if (params?.limit) searchParams.set("limit", String(params.limit));
  if (params?.offset) searchParams.set("offset", String(params.offset));
  const qs = searchParams.toString();
  return get<AuditLogList>(`/v1/teams/${slug}/audit${qs ? `?${qs}` : ""}`);
}

File: api/app/schemas/audit.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class AuditLogResponse(BaseModel):
    id: int
    timestamp: datetime
    user_email: str = Field(..., alias="userEmail")
    action: str
    resource_type: str = Field(..., alias="resourceType")
    resource_id: str | None = Field(None, alias="resourceId")
    team_slug: str | None = Field(None, alias="teamSlug")
    request_body: dict | None = Field(None, alias="requestBody")
    response_status: int | None = Field(None, alias="responseStatus")
    model_config = ConfigDict(from_attributes=True, populate_by_name=True, by_alias=True)


class AuditLogList(BaseModel):
    entries: list[AuditLogResponse]
    total: int
    limit: int
    offset: int
